### PR TITLE
Add Windows System folders monitoring test support

### PR DESCRIPTION
**Pull request by Deblintrake09**
|Related issue|
|-------------|
| #3652            |

## Description

This PR aims to add IT support for FIM monitoring of windows' system folders (System32, SysWOW64) and path redirection from Sysnative to System32.

### Added

- The `test_windows_system_folder_redirection` test suite adds a total of 9 test cases, monitoring the different system folders in each fim_mode.

## Testing performed

| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @Deblintrake09  (Developer)  | test_fim          | [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/35868/)[:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/35869/)[:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/35870/) | ⚫⚫⚫ | Manager        | https://github.com/wazuh/wazuh-qa/pull/3720/commits/235a5b5fdab8a01e717d8bd6f03955153c553049       | Nothing to highlight |
| @Deblintrake09  (Developer)  |  test_fim         | [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/35865/)[:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/35866/)[:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/35867/) | ⚫⚫⚫ | Agent Centos        | https://github.com/wazuh/wazuh-qa/pull/3720/commits/235a5b5fdab8a01e717d8bd6f03955153c553049       | Nothing to highlight |
| @Deblintrake09  (Developer)  |  test_fim         | [:large_blue_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/35387//)[:large_blue_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/35286/)[:large_blue_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/35287/) | ⚫⚫⚫ | Agent  Solaris, macOS        | https://github.com/wazuh/wazuh-qa/pull/3720/commits/c7094996fbc6c591dc29b96dd3d5b991d02925e0       | Cannot launch test on this OSs because of Jenkins Issues |
| @Deblintrake09  (Developer)  |  test_fim    | [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/35922/)[:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/35923/)[:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/35924/) | [:green_circle:](https://github.com/wazuh/wazuh-qa/files/10392083/R1.zip)[:green_circle:](https://github.com/wazuh/wazuh-qa/files/10392084/R2.zip)[:green_circle:](https://github.com/wazuh/wazuh-qa/files/10392085/R3.zip) | Windows   | https://github.com/wazuh/wazuh-qa/pull/3720/commits/235a5b5fdab8a01e717d8bd6f03955153c553049       | Locally launched only added tests |
| @damarisg  (Reviewer)   | /test_fim   | [🟢](https://ci.wazuh.info/view/Tests/job/Test_integration/35360/)  | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |  Manager  |     https://github.com/wazuh/wazuh-qa/pull/3720/commits/9a9efb653e23183f3cbac08d3c835cff813308d4     | Nothing to highlight |
| @damarisg  (Reviewer)   | /test_fim   | [🟢](https://ci.wazuh.info/view/Tests/job/Test_integration/35407/)  | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |  Agent Windows | https://github.com/wazuh/wazuh-qa/pull/3720/commits/9a9efb653e23183f3cbac08d3c835cff813308d4 | Nothing to highlight |

> DEVELOPER NOTE: Local executions only contain added test, because local environment is having problems with running the complete `test_fim` folder. Only ran on Windows since the test has a win32 mark and is automatically skipped in other OSs.

### CODE REVIEW

File: tests/integration/test_fim/test_files/test_windows_system_folder_redirection/test_windows_system_folder_redirection.py
Comment (damarisg) [inline tests/integration/test_fim/test_files/test_windows_system_folder_redirection/test_windows_system_folder_redirection.py:89]: FIM has a lot of tests, so I have a few suggestions:

`1`. To organize the tests correctly, could you improve the path where is store the test? I see words that a lot are repeated.
 
It should be more general to help us separate by sections.

`2`. I would like to know is what happens in this case of synchronization.

We have a section that is to add synchronization tests and contains other cases to check it. Why did you decide not to do synchronization cases?

**Review by damarisg — CHANGES_REQUESTED** [review]

File: tests/integration/test_fim/test_files/test_windows_system_folder_redirection/test_windows_system_folder_redirection.py
Comment (Deblintrake09) [reply tests/integration/test_fim/test_files/test_windows_system_folder_redirection/test_windows_system_folder_redirection.py]: Added in https://github.com/wazuh/wazuh-qa/pull/3720/commits/0712feadcd678d334b1fb42ecdd91a943ae2c888

File: tests/integration/test_fim/test_files/test_windows_system_folder_redirection/test_windows_system_folder_redirection.py
Comment (Deblintrake09) [reply tests/integration/test_fim/test_files/test_windows_system_folder_redirection/test_windows_system_folder_redirection.py]: Fixed in https://github.com/wazuh/wazuh-qa/pull/3720/commits/0712feadcd678d334b1fb42ecdd91a943ae2c888

File: tests/integration/test_fim/test_files/test_windows_system_folder_redirection/test_windows_system_folder_redirection.py
Comment (Deblintrake09) [reply tests/integration/test_fim/test_files/test_windows_system_folder_redirection/test_windows_system_folder_redirection.py]: Fixed in https://github.com/wazuh/wazuh-qa/pull/3720/commits/0712feadcd678d334b1fb42ecdd91a943ae2c888

File: tests/integration/test_fim/test_files/test_windows_system_folder_redirection/data/test_cases/cases_windows_system_folder_redirection.yaml
Comment (Deblintrake09) [reply tests/integration/test_fim/test_files/test_windows_system_folder_redirection/data/test_cases/cases_windows_system_folder_redirection.yaml]: Fixed in https://github.com/wazuh/wazuh-qa/pull/3720/commits/70b42c5edc0523769f83804fe00488943042f686

File: tests/integration/test_fim/test_files/test_windows_system_folder_redirection/test_windows_system_folder_redirection.py
Comment (Deblintrake09) [reply tests/integration/test_fim/test_files/test_windows_system_folder_redirection/test_windows_system_folder_redirection.py]: 1. Renamed the test name to avoid repeating words and be a bit more descriptive.
2. This has not been tested for synchronization, because it is not related to it, no changes related to synchronization. It was needed to test that the folders are monitored and redirection from `Sysnative ` works properly, but once it is monitored, the synchronization behavior is the same as any other folder.

File: tests/integration/test_fim/test_files/test_windows_system_folder_redirection/test_windows_system_folder_redirection.py
Comment (Deblintrake09) [reply tests/integration/test_fim/test_files/test_windows_system_folder_redirection/test_windows_system_folder_redirection.py:32]: This should support all versions of Windows that feature Sytem32, SysWOW64 and Sysnative folders. Only added the OS versions that were used in local testing and on Jenkins.

Should I add the rest?

Comment (damarisg) [reply tests/integration/test_fim/test_files/test_windows_system_folder_redirection/test_windows_system_folder_redirection.py:32]: For now, it's correct. In the future, we will add it and we will test it.

File: tests/integration/test_fim/test_files/test_windows_system_folder_redirection/test_windows_system_folder_redirection.py
Comment (damarisg) [reply tests/integration/test_fim/test_files/test_windows_system_folder_redirection/test_windows_system_folder_redirection.py]: 1. We will wait for another reviewer to review it. I think that it should be improved.

2. Resolve

**Review by damarisg — DISMISSED** [review]

File: tests/integration/test_fim/test_files/test_windows_system_folder_redirection/test_windows_system_folder_redirection.py
Comment (jmv74211) [inline tests/integration/test_fim/test_files/test_windows_system_folder_redirection/test_windows_system_folder_redirection.py:150]: Add comments that clarify what is that function checking or doing

**Review by jmv74211 — CHANGES_REQUESTED** [review]

File: tests/integration/test_fim/test_files/test_windows_system_folder_redirection/test_windows_system_folder_redirection.py
Comment (Deblintrake09) [reply tests/integration/test_fim/test_files/test_windows_system_folder_redirection/test_windows_system_folder_redirection.py:163]: Done in https://github.com/wazuh/wazuh-qa/pull/3720/commits/371a104d1ef726d52e64336a2da80b3a3db63b72

File: tests/integration/test_fim/test_files/test_windows_system_folder_redirection/test_windows_system_folder_redirection.py
Comment (Deblintrake09) [reply tests/integration/test_fim/test_files/test_windows_system_folder_redirection/test_windows_system_folder_redirection.py]: Regarding 2, as talked in call, This fix does not affect synchronization at all, it is just a fix so Windows system folders can be properly monitored. Once monitored, the synchronization works as with any other folder.

**Review by jmv74211 — APPROVED** [review]

**Pull request merged**
